run/make_init_base.py

```diff
@@ -63,140 +46,459 @@ def __init__(self, fname, nFirst, nSecond, res, nCg=1, nDg=1, nDgStress=3, nCoor
+                                lonVerts[ii, jj] += 360
+
+                # Find the centre points along the x-axis
+                # The array looks like this:
+                # [0,0] --- (lat0,lon0) --- [0,1]
+                #   |                         |
+                #   |                         |
+                #   |                         |
+                # [1,0] --- (lat1,lon1) --- [1,1]
+                # Where (lat1,lon1) and (lat0,lon0) are the centre points of the top and bottom sides of the grid cell. I've
+                # removed the division by two from the averaging, as it gets cancelled out in atan2.
+                lat0 = latVerts[0, 0] + latVerts[0, 1]
+                lat1 = latVerts[1, 0] + latVerts[1, 1]
+                lon0 = lonVerts[0, 0] + lonVerts[0, 1]
+                lon1 = lonVerts[1, 0] + lonVerts[1, 1]
+
+                # Calculate the angle as the change in lat/lat coordinates across the grid cell
+                dx = lat1 - lat0
+                dy = lon1 - lon0
+
+                # Negative of the angle to get the rotation of the vector to the displaced pole (not the other way around).
+                # Negative on dy, because the y-axis is pointing downwards.
+                theta[i, j] = -np.arctan2(-dy, dx)
+
+        return theta
+
+    def get_element_longitude(self):
+        """
+        Returns the longitude of the grid cell centre.
+
+        :return: Longitude of the grid cell centre.
+        """
+        return self.__plon
+
+    def get_element_latitude(self):
+        """
+        Returns the latitude of the grid cell centre.
+
+        :return: Latitude of the grid cell centre.
+        """
+        return self.__plat
+
+    def __testFields__(self):
+        """
+        Check if arrays are non-zero and the right size.
+
+        Print a warning if they're zero (this may be ok). Raise a RuntimeError if the
+        shape is wrong.
+        """
+        if (self.mask == 0).all():
+            print("Error: 'mask' is not set (all values are zero, meaning land everywhere)")
+            runtime_err = "'mask' is not set"
+            raise RuntimeError(runtime_err)
+
+        for check in [["cice", (self.cice == 0).all(), self.cice.shape == (self.__nFirst, self.__nSecond)],
+                      ["hice", (self.hice == 0).all(), self.hice.shape == (self.__nFirst, self.__nSecond)],
+                      ["hsnow", (self.hsnow == 0).all(), self.hsnow.shape == (self.__nFirst, self.__nSecond)],
+                      ["sss", (self.sss == 0).all(), self.sss.shape == (self.__nFirst, self.__nSecond)],
+                      ["sst", (self.sst == 0).all(), self.sst.shape == (self.__nFirst, self.__nSecond)]]:
+
+            if self.__checkZeros and check[1]:
+                print("Warning: '" + check[0] + "' is all zeros (this may be ok, if that's what you want).")
+
+            if not check[2]:
+                print("Error: '" + check[0] + "' is the wrong shape")
```

Comment:
This would be more readable with
```suggestion
        for (field, allZero, wrongShape) in [["cice", (self.cice == 0).all(), self.cice.shape == (self.__nFirst, self.__nSecond)],
                      ["hice", (self.hice == 0).all(), self.hice.shape == (self.__nFirst, self.__nSecond)],
                      ["hsnow", (self.hsnow == 0).all(), self.hsnow.shape == (self.__nFirst, self.__nSecond)],
                      ["sss", (self.sss == 0).all(), self.sss.shape == (self.__nFirst, self.__nSecond)],
                      ["sst", (self.sst == 0).all(), self.sst.shape == (self.__nFirst, self.__nSecond)]]:

            if self.__checkZeros and allZero:
                print(f"Warning: '{field}' is all zeros (this may be ok, if that's what you want).")

            if not wrongShape:
                print(f"Error: '{field}' is the wrong shape")
```